### Question 6
We are given the equation Y(t) = α + βX(t) + ε(t) for t = 1 to 20

According to the question, the intercept (α) stays the same, but the slope (β) changed.
A structural break in β at t = 10, means that we have the following equations
- For t = 1 to 10: Y(t) = α + β₁X(t) + ε(t)
- For t = 11 to 20: Y(t) = α + β₂X(t) + ε(t) -- Where β₂ = β₁ + Δβ


To test for a change in slope using regression, (for simplicity) we assume slope and intercept values for the equation before the structural break (say β₁ = 2 and α = 5)

So: β₂ = 2 + Δβ. 

Also for simplicity (assume Δβ = 2), β₂ = 2 + 2 = 4
- For t = 1 to 10: Y(t) = 5 + 2X(t) + ε(t)
- For t = 11 to 20: Y(t) = 5 + 4X(t) + ε(t)

Next, we create a dummy variable D
- D = 0 if t <= 10
- D = 1 if t > 10

Then, we create an additional variable involving X and D: DX = D * X 

Then, we run the regression: 
Y = α + β₁ * X + β₂ * DX + ε (or Y = α + β₁ * X + (β₁ + Δβ) * DX + ε

Lastly, we test for structural break:
- If Δβ is significant (p < 0.05), there's evidence the slope changed after t = 10
- Otherwise, there is no evidence of structural break in the slope

In [12]:
np.random.seed(42)
n = 20
t = np.arange(1, n + 1)

X = np.random.normal(loc = 10, scale = 2, size=n).reshape(-1, 1) # Generate X

a = np.random.uniform(0, 2) # intercept
B = np.random.uniform(1, 3) # slope
b_delta = np.random.uniform(0.5, 2) # change in slope

Y = np.full(20, 0.0) # Initialize Y
for i in range(n):
    if t[i] <= 10:
        Y[i] = a + B * X[i, 0] + np.random.normal(scale = 1)
    else:
        Y[i] = a + (B + b_delta) * X[i, 0] + np.random.normal(scale = 1) # The slope change after the structural break

D = (t > 10).astype(int).reshape(-1, 1)
DX = D * X 

X_new = np.hstack([X, DX]) # Create input features
X_new = sm.add_constant(X_new)

model = sm.OLS(Y, X_new).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.947
Model:                            OLS   Adj. R-squared:                  0.941
Method:                 Least Squares   F-statistic:                     152.3
Date:                Sat, 14 Jun 2025   Prob (F-statistic):           1.40e-11
Time:                        12:55:57   Log-Likelihood:                -28.731
No. Observations:                  20   AIC:                             63.46
Df Residuals:                      17   BIC:                             66.45
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.8061      1.669      0.483      0.6

**Comment:**
- The p value of Δβ (represented by x2) here is 0.000. This is significant because p < 0.05
- So, there is an evidence the slope changed after t = 10

For this equation, the final interpretation here is:
- Before t = 10: Y = 0.8061 + 2.5927 * X
- After t = 10: Y = 0.8061 + (2.5927 + 0.8435) * X = 0.8061 + 3.4362 * X